scripts/installer.py

```diff
@@ -0,0 +1,161 @@
+    MKCERT_FILENAME = ""
+    CERTIFICATIONS_FOLDER_PATH = os.path.join(THYRA_CONFIG_FOLDER_PATH, "certs")
+
+    def __init__(self):
+        pass
+
+    """
+    Prints the error given in parameter and exits the program
+    """
+    def printErrorAndExit(self, error):
+        logging.error(error)
+        os._exit(-1)
+
+    """
+    Executes the command given in parameter and returns the output of the command
+    """
+    def executeCommand(self, command, shell=False) -> str:
+        logging.debug(f'Executing command: {command}')
+        try:
+            process = subprocess.Popen(command, shell=shell,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE
+            )
+            stdout, stderr = process.communicate()
+
+            if stdout is not None and len(stdout) > 0:
+                logging.debug(f'Command output: {stdout.decode(locale.getpreferredencoding(False))}')
+
+            if process.returncode != 0:
+                self.printErrorAndExit(f"Command failed with error : {stderr.decode(locale.getpreferredencoding(False))}")
+
+            return stdout.decode(locale.getpreferredencoding(False))
+        except OSError as err:
+            self.printErrorAndExit(f"Error while executing command: {err}")
+
+    """
+    Downloads the file at the given url and saves it to the given filename
+    """
+    def downloadFile(self, url, filename):
+        logging.debug(f'Downloading {filename} from {url}')
+        try:
+            sslContext = None
+            if platform.system() != "Windows":
+                import certifi
+                sslContext = ssl.create_default_context(cafile=certifi.where()) 
+            with urllib.request.urlopen(url, context=sslContext) as response, open(filename, 'wb') as out_file:
+                shutil.copyfileobj(response, out_file)
+        except URLError as err:
+            logging.info("Failed to download " + filename + " :")
+            self.printErrorAndExit(err)
+        logging.info(filename + " downloaded successfully")
+
+    """
+    This method must be implemented by the installer of each platform to install and configure the DNS server.
+    """
+    def setupDNS(self):
+        pass
+
+    """
+    Generates an HTTPS certificate for my.massa using mkcert and stores it in the thyra config folder.
+    """
+    def generateCertificate(self):
+        logging.info("Generating HTTPS certificate for my.massa")
+
+        if not os.path.exists(self.CERTIFICATIONS_FOLDER_PATH):
+            try:
+                os.makedirs(self.CERTIFICATIONS_FOLDER_PATH)
+            except OSError as err:
+                self.printErrorAndExit(f"Error while creating certs folder: {err}")
+
+        self.downloadFile(self.MKCERT_URL, self.MKCERT_FILENAME)
+        os.chmod(self.MKCERT_FILENAME, 0o755)
+
+        self.executeCommand([
+            os.path.join(os.getcwd(), self.MKCERT_FILENAME), 
+            "--install"])
+        self.executeCommand([
```

Comment:
By experience mkcert is very verbose but share some information on compatibility or action to perform after installation, such as restart the browser for instance. By using this command, will the user get all those messages?

Reply:
No. But `executeCommand` returns the standard output, so we can display it.

Reply:
This is a good question. I think we should have the log:
- in the terminal
- in a dedicated file that can be shared in cased of installation failure

Reply:
I already made some changes related to logs.
Right now, we have logs in the terminal and in a dedicated file `thyra_installer.log`. Those logs are exactly the same.
See https://github.com/massalabs/thyra/blob/c707eeda05627cf52deda47329d291441b4b7e37/scripts/installer.py#L11

About the output of mkcert Install, since `executeCommand` returns the standard output, I can print the result of the function on the standard output (not in the logs) so the user will be noticed about what happened. Or, I can print it using the logger so the user can view it in the installer logs if needed.